scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -257,6 +258,17 @@ object Execution {
   def withConfig[T](ex: Execution[T])(c: Config => Config): Execution[T] =
     TransformedConfig(ex, c)
 
+  def withCached[T](path: String)(fn: CachedFile => Execution[T]): Execution[T] = {
+    Execution.getConfigMode.flatMap { case (config, mode) =>
+      val cachedFile = DistributedCacheFile(path)(mode)
```

Comment:
can we add a comment that this does have the side effect of adding a key to the Configuration, but that should not matter.

On the other hand, maybe we can refactor:
https://github.com/twitter/scalding/blob/develop/scalding-core/src/main/scala/com/twitter/scalding/filecache/DistributedCacheFile.scala#L72

to have a `private[scalding]` method that returns `def immutableApply(path: String)(implicit mode: Mode): (() => Mode, CachedFile)` where the funtion in the left is the side-effect of updating the `Configuration`. The current code would then just do:
```scala
def apply(str: String)(implicit m: Mode): CachedFile = {
  val (effect, cf) = immutableApply(str)
  effect()
  cf
}
```

Then our code can just discard the effect. How does that sound?